tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/json/JsonToTlaViaBuilder.scala

```diff
@@ -0,0 +1,274 @@
+package at.forsyte.apalache.tla.typecomp.json
+
+import at.forsyte.apalache.io.json._
+import at.forsyte.apalache.io.lir.TypeTagReader
+import at.forsyte.apalache.tla.imp.src.{SourceLocation, SourceStore}
+import at.forsyte.apalache.tla.types.tla
+import at.forsyte.apalache.tla.typecomp._
+import at.forsyte.apalache.tla.lir.TypedPredefs._
+import at.forsyte.apalache.tla.lir.src.{SourceLocation, SourcePosition, SourceRegion}
+import at.forsyte.apalache.tla.lir._
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ * A JsonDecoder, which uses the ScopedBuilder to construct TLA IR
+ *
+ * TODO: Once package refactoring is done, this should replace io.json.JsonToTla. Currently impossible due to dependency
+ * issues.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class JsonToTlaViaBuilder[T <: JsonRepresentation](
+    scalaFactory: ScalaFactory[T],
+    sourceStoreOpt: Option[SourceStore] = None,
+  )(implicit typeTagReader: TypeTagReader)
+    extends JsonDecoder[T] {
+  private def missingField(fieldName: String): JsonDeserializationError =
+    new JsonDeserializationError(s"JSON has no field named [$fieldName].")
```

Comment:
The name `missingField` is not very descriptive.
It's also only used once, so I'd either inline, or place the throw inside this method and rename to `throwWithMissingField`.

Reply:
You're right, I just copied this without changing it. IDK why it was originally this way.